Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -328,6 +329,12 @@ def bump_formula_pr
       alias_rename.map! { |a| formula.tap.alias_dir/a }
     end
 
+    ohai "brew update-python-resources #{formula.name}"
+    if !args.dry_run? || args.dry_run? && args.write?
+      PyPI.ensure_pipgrip_installed
```

Comment:
Any reason to not have this called by `PyPI.update_python_resources!`? There don't seem to be any callers that don't need to call this before.

Reply:
The reason I didn't put it in was that when running `brew update-python-resources` with multiple formulae, `PyPI.update_python_resources!` is called in a loop for each formula. It would, therefore, check for `pipgrip` again for each formula which is unecessary.

However, I'm not sure it will save any significant amount of time (plus, I don't know how often people will run with multiple formulae) so it's probably worth adding it to `PyPI.update_python_resources!`

Reply:
I thought that might come up! In that case the result could be memoized so you don't need to query it every time.

Reply:
Not sure what the best way to do this is. Maybe something like this?

```ruby
module PyPI

  @pipgrip_installed = nil

  def update_python_resources!(...)

    @pipgrip_installed ||= Formula["pipgrip"].any_version_installed?
    odie '"pipgrip" must be installed (`brew install pipgrip`)' unless @pipgrip_installed

    ...

  end
end
```

Reply:
This seemed to work correctly locally so I'm going to go with this.

Reply:
Seems fine to me 👍🏻